### PR TITLE
Fix Clippy lint issues for Rust 1.77

### PR DESCRIPTION
**Pull request by Malax**

### CODE REVIEW

File: buildpacks/maven/src/framework.rs [hunk 1]
```diff
@@ -97,12 +97,14 @@ pub(crate) fn default_app_process<P: AsRef<Path>>(
 }
 
 #[derive(Debug)]
+#[allow(dead_code)]
```
Comment (colincasey) [inline buildpacks/maven/src/framework.rs:100]: Instead of disabling the lint here, wouldn't the better option be to take it's advice and drop the inner error if it's not being used anywhere?

File: buildpacks/maven/src/framework.rs [hunk 2]
```diff
@@ -97,12 +97,14 @@ pub(crate) fn default_app_process<P: AsRef<Path>>(
...
 pub(crate) enum DefaultAppProcessError {
     DetectFrameworkError(DetectFrameworkError),
     IoError(std::io::Error),
 }
 
 #[derive(Debug)]
+#[allow(dead_code)]
```
Comment (colincasey) [inline buildpacks/maven/src/framework.rs:107]: Instead of disabling the lint here, wouldn't the better option be to take it's advice and drop the inner error if it's not being used anywhere?

File: buildpacks/maven/tests/integration/versions.rs [hunk 3]
```diff
@@ -115,6 +115,7 @@ fn set_maven_version_app_dir_preprocessor(version: &str, path: &Path) {
         .read(true)
         .write(true)
         .create(true)
+        .truncate(false)
```
Comment (colincasey) [inline buildpacks/maven/tests/integration/versions.rs:118]: I didn't see this in the list of lint errors, why is this changed?

File: buildpacks/maven/src/framework.rs (hunk 1 above)
Comment (Malax) [reply buildpacks/maven/src/framework.rs:100]: It is being used, but only implicitly. The `Debug` implementation is omitted from the dead code check - but we use that to display the inner error should this error occur.

File: buildpacks/maven/src/framework.rs (hunk 2 above)
Comment (Malax) [reply buildpacks/maven/src/framework.rs:107]: It is being used, but only implicitly. The `Debug` implementation is omitted from the dead code check - but we use that to display the inner error should this error occur.

File: buildpacks/maven/tests/integration/versions.rs (hunk 3 above)
Comment (Malax) [reply buildpacks/maven/tests/integration/versions.rs:118]: This is the lint that caused this change: https://rust-lang.github.io/rust-clippy/master/index.html#/suspicious_open_options

**Review by colincasey — APPROVED** [review]
